napi_js_props.md

```diff
@@ -0,0 +1,283 @@
+# N-API - JavaScript Data Types and Values 
+
+N-API exposes a set of APIs to interrogate and set properties on JavaScript objects.
+Some of these types are documented under [Section 7](https://tc39.github.io/ecma262/#sec-operations-on-objects) 
+of the [ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+
+Fundamentally, all property keys in N-API can be represented in one of the following forms:
+- Named: a simple utf8 encoded string
+- Integer-Indexed: an index value represented by `uint32_t`
+- JavaScript Object: a JavaScript Object represented by `napi_value`. This can be a `napi_value` representing a String, Number or Symbol.
+
+N-API values are represented by the type `napi_value`.
+Any N-API call that requires a JavaScript object takes in a `napi_value`.
+However, it's the caller's responsibility to make sure that the 
+`napi_value` in question is of the JavaScript type expected by the API.
+
```

Comment:
Not sure if this is the right place, but I think some explanation and possibly and example of setting attributes up front might make sense (either here or somewhere else in the doc).  The descriptions do a good job of documenting the individual functions but I think this doc should provide a bit more information on how you use them to get things done.